### DISCUSSION TITLE
About How Memory Usage Works

### DISCUSSION THREAD

**Discussion by ghost** [mienaiyami/yomikiru#20] [Q&A] [answered]
**Folder 1**
- Image Type: webp
- Each Image Dimension: 2400 x width varies
- Total Images: 58
- Folder Size: 28 MB

**Folder 2**
- Image Type: jpeg
- Each Image Dimension:  800 x width varies
- Total Images: 117
- Folder Size: 7 MB

Ideally folder 2 should take less memory than folder 1, but it is reverse here.
Is it because this framework works like that?

**Comment by mienaiyami** [comment, marked as answer]
>Is it because this framework works like that?

Yes, you can say that. In almost all framework images are loaded and then rendered, and the pixel data goes to RAM.
Also since webp is recommended format for web, and this framework is like a browser itself, so maybe webp takes less RAM.

**Comment by ghost** [comment]
You can also reproduce the same behavior by following the below steps.

1. Open the folder that you downloaded to check stuttering issue.
2. After opening the folder, minimize/keep the reader idle for 10 - 30 minutes and do your work.
3. After that you can observe the below red color usage.
- **Green Color -** Opened and browsed through immediately.
- **Red Color -** Keeping the reader idle for 20 minutes after loading the manhwa.

### Usage of manga reader after 20 minutes of keeping it idle after loading the manhwa.
![Screenshot](https://user-images.githubusercontent.com/111482404/202863098-317e3070-ca9e-4f7b-b379-75456ecfd082.jpg)

- Even after scrolling to the bottom of the manhwa the usage remained the same at 28 - 30 MB for red color.
- Normally using it also takes less ram 60 - 70 Mb.
- Both performed the same, but the red color ram usage is 50% of green color with the same performance.

**Reply by mienaiyami** [reply]
I will be honest; I have no idea how electronJS works. RAM usage is about 44-50 when app is empty, with no code. I have tried searching but found no results (when I started building this app). Soon i plan to try coding same app in "Tauri". 
Check this if you are interested https://github.com/Elanis/web-to-desktop-framework-comparison/tree/main.
I can only make UI using HTML, CSS, JS so i need to rely on this framework. 
But take a look at build time of Tauri, that scares me but I will try building in it either way.

**Comment by ghost** [comment]
> Check this if you are interested https://github.com/Elanis/web-to-desktop-framework-comparison/tree/main.

Thanks for the link. I checked it out.

From what I understood, Tauri is the clear winner as Tauri takes way less RAM, app size as compared to electron and others.
1. Build size of the same app in, Tauri - 3 MB and Electron - 226MB.
2. RAM usage is <50% as compared to electron in release state.

>But take a look at build time of Tauri, that scares me but I will try building in it either way.

- Compilation took 8 to 9 minutes in Tauri, but for the same app, electron took 4 to 5 seconds.
- **The build time is the only drawback of Tauri.**
- **I look forward to the same app in Tauri.**

**Reply by mienaiyami** [reply]
>Compilation took 8 to 9 minutes in Tauri, but for the same app, electron took 4 to 5 seconds.

Thats not actual time to compile. It takes 2min to compile my app and make it 4 if you include portable version as well, and that's on an SSD.

But the thing with tauri is, it uses a different language and a complex one at that. Though I plan to learn it, I wouldn't be able to master it like JS which I have been using for 4 years. And after that's comes features problem, for example, it doesn't have option to disable hardware acceleration, which will make app stutter while scrolling images.

But nonetheless I will try my hand at it soon.

**Comment by ghost** [comment]
Finally, tauri has given an option to '[disable hardware acceleration](https://github.com/tauri-apps/wry/issues/625)' for windows users now.
